WD-Firmware/examples/build/SConscript_rtosal

```diff
@@ -15,47 +15,45 @@
  limitations under the License.
 '''
 import os
+import utils
 Import('Env')
 
 out_dir = os.path.join(Env['OUT_DIR_PATH'], 'rtosal')
-if not os.path.exists(out_dir):
-    os.makedirs(out_dir)
+utils.createFolder(out_dir)
 
 rtosal_base = os.path.join('rtos', 'rtosal')
 list_files_c=[
-   [os.path.join(rtosal_base,'rtosal_event.c'), os.path.join(out_dir, 'rtosal_event.o')],
-   [os.path.join(rtosal_base,'rtosal_util.c'), os.path.join(out_dir, 'rtosal_util.o')],
-   [os.path.join(rtosal_base,'rtosal_mutex.c'), os.path.join(out_dir, 'rtosal_mutex.o')],
-   [os.path.join(rtosal_base,'rtosal_queue.c'), os.path.join(out_dir, 'rtosal_queue.o')],
-   [os.path.join(rtosal_base,'rtosal_semaphore.c'), os.path.join(out_dir, 'rtosal_semaphore.o')],
-   [os.path.join(rtosal_base,'rtosal_task.c'), os.path.join(out_dir, 'rtosal_task.o')],
-   [os.path.join(rtosal_base,'rtosal_time.c'), os.path.join(out_dir, 'rtosal_time.o')],
-   [os.path.join(rtosal_base,'rtosal_default.c'), os.path.join(out_dir, 'rtosal_default.o')],
-#   [os.path.join(rtosal_base,'rtosal_memory.c'), os.path.join(out_dir, 'rtosal_memory.o')],
+   [os.path.join(rtosal_base, 'src', 'rtosal_event.c'), os.path.join(out_dir, 'rtosal_event.o')],
+   [os.path.join(rtosal_base, 'src', 'rtosal_util.c'), os.path.join(out_dir, 'rtosal_util.o')],
+   [os.path.join(rtosal_base, 'src', 'rtosal_mutex.c'), os.path.join(out_dir, 'rtosal_mutex.o')],
+   [os.path.join(rtosal_base, 'src', 'rtosal_queue.c'), os.path.join(out_dir, 'rtosal_queue.o')],
+   [os.path.join(rtosal_base, 'src', 'rtosal_semaphore.c'), os.path.join(out_dir, 'rtosal_semaphore.o')],
+   [os.path.join(rtosal_base, 'src', 'rtosal_task.c'), os.path.join(out_dir, 'rtosal_task.o')],
+   [os.path.join(rtosal_base, 'src', 'rtosal_time.c'), os.path.join(out_dir, 'rtosal_time.o')],
+   [os.path.join(rtosal_base, 'src', 'rtosal_error.c'), os.path.join(out_dir, 'rtosal_error.o')],
+   [os.path.join(rtosal_base, 'src', 'rtosal_interrupt.c'), os.path.join(out_dir, 'rtosal_interrupt.o')],
 ]
 
 list_files_a=[
-   [os.path.join(rtosal_base,'rtosal_int_vect.S'), os.path.join(out_dir, 'rtosal_int_vect.o')],
+   [os.path.join(rtosal_base, 'src', 'rtosal_int_vect.S'), os.path.join(out_dir, 'rtosal_int_vect.o')],
 ]
 
-rtos_core_define = {
-  'freertos' :  ['D_USE_FREERTOS'],
-  'threadx'  :  ['D_USE_THREADX'],
-}
-
 # compiler flags
 c_flags = [] + Env['C_FLAGS']
 a_flags = [] + Env['A_FLAGS']
 
 # defines
-Env['RTOSAL_DEF'] = rtos_core_define[Env['RTOS_CORE']]
-c_defines = Env['BSP_DEF'] + Env['RTOSAL_DEF'] # add "+ ['D_RTOSAL_VECT_TABLE']" when using vector table
+c_defines = Env['BSP_DEF'] + Env['RTOSAL_DEF'] + Env['PUBLIC_DEF'] # add "+ ['D_RTOSAL_VECT_TABLE']" when using vector table
 
 # include paths
 local_includes  = [os.path.join(Env['ROOT_DIR'], rtosal_base, 'loc_inc'),]
 public_includes = [os.path.join(Env['ROOT_DIR'], rtosal_base, 'api_inc'),
+                   os.path.join(Env['ROOT_DIR'], rtosal_base, 'config', 'hifive-1'),

```

Comment:
once will go to public then 'target' will be part of the path for the include which goes to public